src/periphery/SimplePurchaseNFT.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import {ERC721TokenReceiver} from "solmate/tokens/ERC721.sol";
+import {ERC20} from "solmate/tokens/ERC20.sol";
+
+import {ExponentialPriceDecayNFTAuction} from "src/ExponentialPriceDecayNFTAuction.sol";
+
+contract SimplePurchaseNFT is ERC721TokenReceiver {
+    /// @notice used to guard onERC721Received
+    /// only has a non 0 value during a transaction
+    /// used to only accept NFTs from the auction contract being called
+    /// in the current tx
+    ExponentialPriceDecayNFTAuction currentAuctionContract;
+
+    error WrongFrom();
+
+    /// @notice Purchases the NFT being sold in auction by auctionContract
+    /// @param auctionContract the ExponentialPriceDecayNFTAuction contract selling the NFT
+    /// @param auction the details of the auction
+    /// @param maxPrice the maximum the caller is willing to pay
+    function purchaseNFT(
+        ExponentialPriceDecayNFTAuction auctionContract,
```

Comment:
why is `auctionContract` passed into `purchaseNFT` rather than being set on the constructor of this contract for example?

Reply:
because there are many possible auction contract addresses